### DISCUSSION TITLE
Virtual Machine IPAM Parent Interface

### DISCUSSION THREAD

**Discussion by Luuke257** [netbox-community/netbox#6390] [Q&A] [answered]
Hey!

I recently ran into some Problems Assigning Interfaces with Virtual Machines in Netbox.
I Want to Assign a Virtual Interface to my  VM and set its Parent Interface to the physical Interface on my hypervisor, marking the Network Card that is shared between the host and the VM, but i cant figure out how. As far as i can see, i can only assign virtual interfaces to the host itself and set its Parent Interface to the physical one, but then its not linked to my VM. When i Assign an Interface to my VM however, i cant set its Parent Interface to my Physical Interface @ my host. 

Am i missing something here? Or is this intended?

Any help would be appreciated, thanks!

**Comment by candlerb** [comment, marked as answer]
It is as intended: you cannot associate an interface on a VM with an interface on a device.  Note that in the Netbox model, a VM is implicitly mobile between all devices in a "cluster".

The main use case for interface parents is subinterfaces: e.g. for interface eth0.100, its parent interface is eth0.

For your application, I think you would be better assigning the VM interface as "access mode" assigned to a specific VLAN.  Then whichever physical port(s) on your VM host(s) that VLAN appears on, are where the traffic will appear.

**Reply by Luuke257** [reply]
Alright, i can see the usecase for that. I just thought of a different way which would have been nicer for me, but hey its not always about being easy. Thanks for the clarification!
